experimental/dds/tree2/src/test/feature-libraries/editable-tree/editableTree.editing.spec.ts

```diff
@@ -662,9 +664,80 @@ describe("editable-tree: editing", () => {
 				field.setContent(undefined);
 			});
 
-			// delete method
+			// remove method
 			assert.throws(() => {
-				field.delete();
+				field.remove();
+			});
+		});
+	});
+
+	describe("treeStatus", () => {
+		describe("EditableTree", () => {
+			it("root node returns TreeStatus.InDocument", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: [], foo2: [] },
+				});
+				const rootNode = view.root;
+				assert(isEditableTree(rootNode));
+				const rootNodeStatus = rootNode[treeStatus]();
+
+				assert.equal(rootNodeStatus, TreeStatus.InDocument);
+			});
+
+			it("removed node returns TreeStatus.Deleted", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: ["foo", "bar"], foo2: [] },
+				});
+				const root = view.root;
+				assert(isEditableTree(root));
+				const field = root[localFieldKey];
+				assert(isEditableField(field));
+
+				const node = field.getNode(0);
+				const statusBeforeRemove = node[treeStatus]();
+				assert.equal(statusBeforeRemove, TreeStatus.InDocument);
+
+				field.remove();
+				const statusAfterRemove = node[treeStatus]();
+				assert.equal(statusAfterRemove, TreeStatus.Deleted);
+			});
+		});
+
+		describe("EditableField", () => {
+			it("root field returns TreeStatus.InDocument", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: [], foo2: [] },
+				});
+				const rootField = view.context.root;
+				const rootNodeStatus = rootField.treeStatus();
+
+				assert.equal(rootNodeStatus, TreeStatus.InDocument);
+			});
+
+			it("removed field returns TreeStatus.Deleted", () => {
+				const view = createSharedTreeView().schematize({
+					schema: getTestSchema(FieldKinds.sequence),
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: { foo: ["foo"], foo2: [] },
```

Comment:
In both the node and field tests, you include an extra child, but don't do anything with it.

You should probably extend the tests so that they ensure the contents of deleted/removed trees are also considered deleted/removed.

You also only test the inDocument case at the root. either extends (and rename) those tests to check non-root cases or add addationl tests for non-root fields and nodes.

While there isn't much special logic for non-root nodes (so maybe just renaming the node test to "node returns TreeStatus.InDocument" is enough for it, as "root" nodes arn't that special, root fields had different code handleing them then non-root fields, so I think they deserve a could extra lines in the test or a separate test.

Reply:
Updated test cases